libraries/WiFiClientSecure/src/WiFiClientSecure.cpp

```diff
@@ -263,6 +265,17 @@ void WiFiClientSecure::setCACert (const char *rootCA)
     _CA_cert = rootCA;
 }
 
+ void WiFiClientSecure::setCACertBundle(const uint8_t * bundle)
+ {
+    if (bundle != NULL)
+    {
+        esp_crt_bundle_set(bundle);
+        _use_ca_bundle = true;
+    } else {
+        _use_ca_bundle = false;
```

Comment:
shouldn't any old bundle be freed here?

Reply:
In the normal usage case, the "bundle" variable points to a static string in flash (see README), so trying to free it would not make sense, and also result in a crash (note that the variable is passed as a const, so it should not be in WiFiClientSecure's contract to free it in any case). Internally esp_crt_bundle_set frees s_crt_bundle.crts, which is the only dynamically allocated element in the mechanism (it is the index created for searching the bundle during authentication).

Reply:
The only reason I am asking is because I see in `esp_crt_bundle_set` you indeed free, but if bundle is `NULL`, `esp_crt_bundle_set` would not be called.

Reply:
You are correct, it makes sense to clean up the old bundle index when a NULL bundle is attached. I added an esp_crt_bundle_detach() call on that branch.